### ISSUE TITLE
v1.0.0-beta.1: AppImage crash & unusable user interface (Linux Mint 21.3, X11)

### ISSUE THREAD

**Issue by Moini** [friction2d/friction#341]
Hi!

The AppImage behaves very oddly on my system.

* the user interface looks incomplete
* when I hover over the menu, items get cut off
* dialogs are cut off
* when I start a new project, except for the buttons on the left side, there is basically no user interface
* when I open the preferences to see whether there's a theme issue, perhaps, I get an instant crash.

https://github.com/user-attachments/assets/3f511a5c-3bc5-4b39-91d4-09f8443a26bf

![grafik](https://github.com/user-attachments/assets/be54d87a-37e2-48b9-ace7-3e007bf589dc)

Command line output (ends with crash on preferences menu choice):

```
$ ./Friction-1.0.0-beta.1-x86_64.AppImage 
Friction 1.0.0-beta.1 - https://friction.graphics
OpenGL Vendor: AMD
OpenGL Renderer: RENOIR (renoir, LLVM 15.0.7, DRM 3.57, 6.8.0-49-generic)
OpenGL Version: 4.6 (Core Profile) Mesa 23.2.1-1ubuntu3.1~22.04.2
---
Entered initializeGpu
OffscreenQGL33c initialized
Make OffscreenQGL33c current
iniPlainVShaderVBO
iniPlainVShaderVAO
iniTexturedVShaderVBO
iniTexturedVShaderVAO
iniColorPrograms
Done OffscreenQGL33c current
Entered GpuExecController initialize
GpuTaskExecutor initialized
Entered GpuTaskExecutor::initializeContext
Created GrGLInterface
Created GrContext
iniTexturedVShaderVAO
SwitchableContext set
Done GpuTaskExecutor::initializeContext
Only C and default locale supported with the posix collation implementation
Only C and default locale supported with the posix collation implementation
Case insensitive sorting unsupported in the posix collation implementation
Numeric mode unsupported in the posix collation implementation
Only C and default locale supported with the posix collation implementation
Only C and default locale supported with the posix collation implementation
Case insensitive sorting unsupported in the posix collation implementation
Numeric mode unsupported in the posix collation implementation
Only C and default locale supported with the posix collation implementation
Only C and default locale supported with the posix collation implementation
Case insensitive sorting unsupported in the posix collation implementation
Numeric mode unsupported in the posix collation implementation
Closing Friction, please wait ... 
```

**Comment by rodlie** [comment]
Broken driver and/or Mesa.

`LIBGL_ALWAYS_SOFTWARE=true ./Friction-1.0.0-beta.1-x86_64.AppImage `

**Comment by Moini** [comment]
I can get it to this, with dragging on the divider and drawing something:

![grafik](https://github.com/user-attachments/assets/f549c81e-402c-4c51-96ac-5627240d735c)


* Pointer highlight in the timeline area is shifted (it is three lines below the actual mouse cursor)

**Comment by rodlie** [comment]
I don't know, not able to replicate *(tried Mint 21.3 LiveISO, worked fine)*.

Note I do not test Radeon cards, only Intel and NVIDIA.

**Comment by Moini** [comment]
Could it be a theme incompatibility? And perhaps an additional issue with the font size (I think I increased it on my system, getting older...).

If it's not that, it looks like it might be the Radeon, then:

```
CPU:
  Info: 6-core model: AMD Ryzen 5 4500U with Radeon Graphics bits: 64 type: MCP smt: <unsupported>
    arch: Zen 2 rev: 1 cache: L1: 384 KiB L2: 3 MiB L3: 8 MiB
  Speed (MHz): avg: 1952 high: 3978 min/max: 1400/2375 boost: enabled cores: 1: 1396 2: 1397
    3: 1396 4: 1774 5: 1774 6: 3978 bogomips: 28444
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm
Graphics:
  Device-1: AMD Renoir vendor: Acer Incorporated ALI driver: amdgpu v: kernel pcie: speed: 16 GT/s
    lanes: 16 ports: active: HDMI-A-1,eDP-1 empty: none bus-ID: 05:00.0 chip-ID: 1002:1636
    class-ID: 0300
  Display: x11 server: X.Org v: 1.21.1.4 compositor: xfwm v: 4.18.0 driver: X:
    loaded: amdgpu,ati unloaded: fbdev,modesetting,vesa gpu: amdgpu display-ID: :0.0 screens: 1
  Screen-1: 0 s-res: 3840x1200 s-dpi: 96 s-size: 1016x317mm (40.0x12.5") s-diag: 1064mm (41.9")
  OpenGL: renderer: RENOIR (renoir LLVM 15.0.7 DRM 3.57 6.8.0-49-generic)
    v: 4.6 Mesa 23.2.1-1ubuntu3.1~22.04.2 direct render: Yes
```

Is there anything I can do to help debug?

**Comment by Moini** [comment]
Update: with the second beta, the libgl command no longer helps to prevent a crash when trying to access preferences. The rest of the user interface still looks the same as before.

**Comment by rodlie** [comment]
> Is there anything I can do to help debug?

* Resolution
* Display scaling
* Font size
* Desktop (gnome, kde etc and version if possible) 

> Update: with the second beta, the libgl command no longer helps to prevent a crash when trying to access preferences. The rest of the user interface still looks the same as before.

Beta 2 has no changes to preferences compared to Beta 1. Also built identical, no build or SDK changes happened between the two.

**Comment by rodlie** [comment]
Note that when you click preferences a scan for audio devices will occur, could be related to your problem.

**Comment by rodlie** [comment]
Did a reinstall of Ubuntu 22.04 (using latest ISO) on a test computer last night (and of course they have changed the kernel again, "LTS quality" strikes again), and I can confirm the same issue.

Kernel 6.8 is broken, don't use it. Downgrading or upgrading the kernel "fixes" the issue.

**Comment by Moini** [comment]
Thanks for testing. I have since updated to LM based on Ubuntu 24.04. It's still using the 6.8 kernel, though. The only other available kernel seems to be 6.11.0, haven't tried that yet, though.
